### DISCUSSION TITLE
Interrupting a progress bar gracefully

### DISCUSSION THREAD

**Discussion by jamesb93** [Textualize/rich#1010] [Q&A]
Hello all, making a small timer work tracker app for myself as my favourite one closed shop. At the moment I have a progress bar as a countdown that I want to be able to pause and unpause. At the moment this works but is pretty messy and results in several reprints of the bar to the terminal when keyboard interrupt is invoked.

```python
@cli.command()
@click.argument('project_name')
def start(project_name: str):
    project_exists(project_name)
    with Progress() as progress:
        work = progress.add_task("[red]Work!", total=WORK_PERIOD)

        while not progress.finished:
            try:
                tick = 1
                progress.update(work, advance=tick)
                time.sleep(tick)
            except KeyboardInterrupt:
                input('Paused')
    project_update(project_name)
```

Is there a way, perhaps similar to what I am doing now that anyone has come across or implemented themselves for stopping the progress bar temporarily?

**Comment by willmcgugan** [comment]
Using input will echo to the screen and break the display of Live. Rich doesn't have any other keyboard support, but you could use the `pynput` module on Pypi.
